atomic_reactor/plugins/post_import_image.py

```diff
@@ -88,19 +88,28 @@ def run(self):
                           self.docker_image_repo)
 
             # Tags are imported automatically on creation
-            osbs.create_image_stream(self.imagestream, self.docker_image_repo,
-                                     **kwargs)
-        else:
-            self.log.info("Importing new tags for %s", self.imagestream)
-
-            attempts = 0
-            while not osbs.import_image(self.imagestream):
-                attempts += 1
-
-                if attempts >= self.import_attempts:
-                    msg = "Failed to import new tags for %s"
-                    raise RuntimeError(msg % self.imagestream)
-
-                self.log.info("no new tags, will retry after %d seconds (%d/%d)",
-                              self.retry_delay, attempts, self.import_attempts)
-                sleep(self.retry_delay)
+            i = osbs.create_image_stream(self.imagestream, self.docker_image_repo,
+                                         **kwargs)
+        self.log.info("Importing new tags for %s", self.imagestream)
+
+        primaries = None
+        try:
+            primaries = self.workflow.build_result.annotations['repositories']['primary']
+        except:
+            pass
+        if primaries:
+            for s in primaries:
+                tag_image_name = ImageName.parse(s)
+                osbs.ensure_image_stream_tag(i.json(), tag_image_name.tag)
```

Comment:
In order to get a list of all missing imagestream tags we should be catching `OsbsResponseException` and logging it. If some exceptions were thrown in the loop a `RuntimeError` should be thrown with a message `some imagestreams were not created, check the logs`. This would allow us to get a full list of missing imagestreams, not just the first one.